src/test/java/nextstep/subway/unit/LineServiceMockTest.java

```diff
@@ -1,26 +1,282 @@
 package nextstep.subway.unit;
 
+import nextstep.subway.line.dto.LineRequest;
+import nextstep.subway.line.dto.LineResponse;
+import nextstep.subway.line.dto.LineUpdateRequest;
+import nextstep.subway.line.dto.SectionRequest;
+import nextstep.subway.line.entity.Line;
+import nextstep.subway.line.entity.Section;
+import nextstep.subway.line.entity.Sections;
+import nextstep.subway.line.repository.LineRepository;
+import nextstep.subway.line.service.LineService;
+import nextstep.subway.station.entity.Station;
+import nextstep.subway.station.repository.StationRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 
+import javax.persistence.EntityNotFoundException;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.*;
+
 @ExtendWith(MockitoExtension.class)
 public class LineServiceMockTest {
+
     @Mock
     private LineRepository lineRepository;
+
     @Mock
-    private StationService stationService;
+    private StationRepository stationRepository;
+
+    @InjectMocks
+    private LineService lineService;
+
+    private Station 강남역;
+    private Station 역삼역;
+    private Line 신분당선;
+
+    @BeforeEach
+    void setUp() {
+        강남역 = new Station("강남역");
+        역삼역 = new Station("역삼역");
+        신분당선 = new Line("신분당선", "bg-red-600", 강남역, 역삼역, 10);
+    }
+
+    @DisplayName("지하철 노선을 생성한다.")
+    @Test
+    void 지하철_노선_생성() {
+        // given
+        final LineRequest lineRequest = this.신분당선_생성_요청();
+
+        when(stationRepository.findById(1L)).thenReturn(Optional.of(강남역));
+        when(stationRepository.findById(2L)).thenReturn(Optional.of(역삼역));
+        when(lineRepository.save(any(Line.class))).thenReturn(신분당선);
+
+        // when
+        final LineResponse response = lineService.createSubwayLine(lineRequest);
+
+        // then
+        assertThat(response.getName()).isEqualTo(lineRequest.getName());
+        assertThat(response.getColor()).isEqualTo(lineRequest.getColor());
+        verify(lineRepository).save(any(Line.class));
```

Comment:
```suggestion
        verify(lineRepository, times(1)).save(any(Line.class));
```
이런 식으로 몇번 해당 함수가 호출되었는지 검증해보는 것은 어떤가요?

Reply:
verify를 처음 사용해봐서 2번 이상 함수 호출되는 것들은 테스트 실패 메시지가 출력되어서 추가했었는데
1번도 정상적으로 1번만 동작하는지 명시적으로 체크하는 것이 확실히 좋겠네요!
앞으로 명시해보겠습니다